### ISSUE TITLE
Loaded Game has dozen's of 'Ride Broken Down' messages for rides that don't exist

### ISSUE THREAD

**Issue by extrobe** [OpenRCT2/OpenRCT2#17600]
### Operating System

MacOS 12.5

### OpenRCT2 build

OpenRCT2, v0.4.1 (be518f4 on HEAD)

### Describe the issue

Whilst playing scenario 'Millenium Mines' I saved progress. When I load that saved game, the 'Recent Messages' box shows multiple ' ... has broken down' messages, even though those rides do not, and have not, existed in this park. The dates are also out of sync with when I actually saved the game (June, messages dates March)

![Screen Shot 2022-07-24 at 18 58 13](https://user-images.githubusercontent.com/25456852/180640098-8dc6c9e8-c276-4434-ab58-5239a7bb3965.png)

I then exited the game (without saving), made a copy of the .park file, open _that_ save, and got a _different_ set of messages (this time in French)

![Screen Shot 2022-07-24 at 19 08 29](https://user-images.githubusercontent.com/25456852/180640279-ec6abe04-e34c-4b4d-890b-e7d09256c9e7.png)

### Steps to reproduce

1. Started new scenario
2. Saved progress
3. closed game
4. Started game
5. Loaded saved game
6. checked messages

I've open a few different past saved games, all many of my saved games showed the same behaviour - though not all of them

**Comment by SpartanFrederic104** [comment]
Is this just .park files or are .sv6 and .sv4 files affected too?

**Comment by extrobe** [comment]
I don't have any sv6 files of my own, but downloaded a few and didn't see the same behaviour. (Appreciate that's not super definitive though!)

**Comment by lalip** [comment]
I had this happen just now. Here's my save file; honestly I don't know what led to the bug appearing. I didn't restart the game at any point since starting the scenario, but it spent a couple of hours paused with the computer in sleep mode.
[issue17600.zip](https://github.com/OpenRCT2/OpenRCT2/files/9204901/issue17600.zip)

**Comment by lalip** [comment]
I had time today to take a closer look at this, and I think I've narrowed it down to a reproducible cause. I'm not fluent in production-grade C++ and couldn't figure out the culprit, though.

[Here's a 1:30 video showing the following key findings](https://streamable.com/c45oo1):
- In a nutshell, loading a save with anywhere between 1 and 49 notifications will pad these to exactly 50 with bogus ones.
- Save files generated from and opened in v0.4.0 and v0.4.1 are affected, whereas those generated from v0.3.5.1 are not.
- The generated notifications are extremely consistent between attempts, except for a single event where they showed up in french. Both cases appear identical to the ones shown by @extrobe.

Steps for reproducing this bug:
1. Start a new scenario game
2. Generate at least 1 and at most 49 notifications (I did this by crashing a ride)
3. Save the game
4. Quit and load the game back up (loading in-place does not trigger the bug)

[Here are my save files for convenience](https://github.com/OpenRCT2/OpenRCT2/files/9223783/issue17600-3.zip)
